src/main/java/fi/fmi/avi/model/immutable/TacGeometryImpl.java

```diff
@@ -0,0 +1,67 @@
+package fi.fmi.avi.model.immutable;
+
+import java.io.Serializable;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.inferred.freebuilder.FreeBuilder;
+
+import fi.fmi.avi.model.TacGeometry;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+
+/**
+ * Created by rinne on 18/04/2018.
+ */
+@FreeBuilder
+@JsonInclude(JsonInclude.Include.NON_DEFAULT)
+@JsonDeserialize(builder = TacGeometryImpl.Builder.class)
+public abstract class TacGeometryImpl implements TacGeometry, Serializable {
+
+    private static final long serialVersionUID = 1234L;
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static TacGeometryImpl immutableCopyOf(final TacGeometry tacGeometry) {
+        Objects.requireNonNull(tacGeometry);
+        if (tacGeometry instanceof SurfaceWindImpl) {
+            return (TacGeometryImpl) tacGeometry;
+        } else {
+            return Builder.from(tacGeometry).build();
+        }
+    }
+
+    @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+    public static Optional<TacGeometryImpl> immutableCopyOf(final Optional<TacGeometry> tacGeometry) {
+        Objects.requireNonNull(tacGeometry);
+        return tacGeometry.map(TacGeometryImpl::immutableCopyOf);
+    }
+
+    public abstract Builder toBuilder();
+
+    @SuppressWarnings("EmptyMethod")
+    public static class Builder extends TacGeometryImpl_Builder {
+
+        @Deprecated
+        public Builder() {}
```

Comment:
As a new model class, it would be best not to expose the constructor at all, but make it package-private without annotating it deprecated.

This would also be a good time to deprecate public builder class constructors of remaining SIGMET/AIRMET related model classes that still have a public constructor, and replace them with a static `builder()` factory method. Remember also to deprecate default (undeclared) constructors, which are public by nature